```{r}
library(magrittr)
```

Define parser for GTF attributes

```{r}
GTF_HEADER <- c(
  "seqname",
  "source",
  "feature",
  "start",
  "end",
  "score",
  "strand",
  "frame",
  "attributes"
)

parse_attribute <- function(string) {
  tmp <- string %>%
    trimws(., which = "both") %>%
    strsplit(., ";") %>%
    unlist %>%
    trimws(., which = "both") %>%
    strsplit(., " ")
  out <- c()
  for (i in tmp) {
    out[i[1]] = paste(i[-1], collapse = " ")
  }
  return(out)
}
# import gtf
ginger_gtf <- read.table(
  file = "references/ncbi_dataset/data/GCF_018446385.1/genomic_modified.gtf",
  sep = "\t",
  header = FALSE,
  col.names = GTF_HEADER,
  comment.char = "#"
)

# find every first CDS lines of each protein coding gene
cds_lines <- ginger_gtf$feature == "CDS"
every_first_cds <- 
  (cds_lines & c(FALSE, !cds_lines[-length(cds_lines)])) %>% 
  which

# get protein ID and gene ID from parsed attributes
ginger_p2g <- matrix(
  ncol = 2,
  nrow = length(every_first_cds)
)
j <- 1
for (nth_row in every_first_cds) {
  attrs <- parse_attribute(ginger_gtf$attributes[nth_row])
  ginger_p2g[j, 1] <- attrs["protein_id"]
  ginger_p2g[j, 2] <- attrs["gene_id"]
  j <- j + 1
}

# set row name of the matrix as protein ID
rownames(ginger_p2g) <- ginger_p2g[, 1]
colnames(ginger_p2g) <- c("pid", "gid")

```

2.  *A. thaliana*

```{r}
ath_annot_info <- read.table(
  file = "references/ath/Phytozome/PhytozomeV12/Athaliana/annotation/Athaliana_167_TAIR10.annotation_info.txt",
  header = TRUE,
  sep = "\t",
  comment.char = ""
)
head(ath_annot_info)
ath_p2g <- ath_annot_info[, c("peptideName", "locusName")] %>% as.matrix
colnames(ath_p2g) <- c("pid", "gid")
rownames(ath_p2g) <- ath_p2g[, 1]
```

```{r}
blastp_res <- read.table(
  file = "outputs/blastp/ginger_vs_ath.tsv",
  header = FALSE,
  sep = "\t"
)
blastp_res
```
